editor/src/components/canvas/plugins/tailwind-style-plugin-utils/tailwind-responsive-utils.ts

```diff
@@ -0,0 +1,76 @@
+import type { Config } from 'tailwindcss/types/config'
+import { type StyleMediaSizeModifier, type StyleModifier } from '../../canvas-types'
+import type { ScreenSize } from '../../responsive-types'
+import { extractScreenSizeFromCss } from '../../responsive-utils'
+
+export const TAILWIND_DEFAULT_SCREENS = {
+  sm: '640px',
+  md: '768px',
+  lg: '1024px',
+  xl: '1280px',
+  '2xl': '1536px',
+}
+const defaultTailwindConfig = {
+  theme: {
+    screens: TAILWIND_DEFAULT_SCREENS,
+  },
+} as unknown as Config
+type TailwindScreen = string | { min: string; max: string }
+
+export function screensConfigToScreenSizes(config: Config | null): Record<string, ScreenSize> {
+  const tailwindConfig = config ?? defaultTailwindConfig
+  const screenSizes: Record<string, TailwindScreen> = {
+    ...((tailwindConfig.theme?.screens as Record<string, TailwindScreen>) ??
+      TAILWIND_DEFAULT_SCREENS),
+    ...((tailwindConfig.theme?.extend?.screens as Record<string, TailwindScreen>) ?? {}),
+  }
+
+  return Object.fromEntries(
+    Object.entries(screenSizes)
+      .map(([key, size]) => {
+        const mediaString =
+          typeof size === 'string'
+            ? `@media (min-width: ${size})`
+            : `@media ${[
+                size.min != null ? `(min-width: ${size.min})` : '',
+                size.max != null ? `(max-width: ${size.max})` : '',
+              ]
+                .filter((s) => s != '')
+                .join(' and ')}`
+
+        const screenSize = extractScreenSizeFromCss(mediaString)
```

Comment:
`screenSizes` already comes in the form of min or {min, max}. If I understand this correctly,  we create a proper media query string from it, and then `extractScreenSizeFromCss` strips the min and max and returns it to us. Am I missing some nuance here? feels like unnecessary double work

Reply:
It's made intentionally to decouple the Tailwind specific code from the generic media query parsing code. This two-steps process allows us to use the generic part for any media query parsing (even if it didn't come from Tailwind), and also be tested separately and self-contained